odoo/models.py

```diff
@@ -1698,6 +1699,97 @@ def _read_group_fill_results(self, domain, groupby, remaining_groupbys,
 
         return list(result.values())
 
+    @api.model
+    def _read_group_fill_temporal(self, data, groupby, aggregated_fields,
+                                  interval=dateutil.relativedelta.relativedelta(months=1)):
+        """Helper method for filling dates 'holes' in a result set.
+
+        We are in a use case where data are grouped by a date field (typically
+        months but it could a another interval) and displayed in a Bar Chart or
+        a Line Chart.
+
+        Let's says a request has to group records by month for August, September
+        and ...December.  If we don't changed anything, we would get a Bar Chart
+        looking like this :
```

Comment:
Assume we group records by month, and we only have data for August, September and December. By default, plotting the result gives something like: